File: Dragons.py
# s, n = map(int, input().split())

# l = []
# for i in range(n):
#     x, y = map(int, input().split())
#     l.append([x, y])

# print(l)


# s, n = map(int, input().split())
# l = [[int(x) for x in input().split()] for _ in range(n)]
# print(l)

s, n = map(int, input().split())
l = [list(map(int, input().split())) for _ in range(n)]
# print(l)

l.sort(key=lambda x: x[0])
c = 0

for dragon in l:
    x, y = dragon
    if x < s:
        c += 1
        s += y
    else:
        break

if c == n:
    print("YES")
else:
    print("NO")

File: Dragons.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int s, n;
    cin >> s >> n;

    vector<pair<int, int>> dragons(n);
    for (int i = 0; i < n; i++) {
        int x, y;
        cin >> x >> y;
        dragons[i] = make_pair(x, y);
    }

    sort(dragons.begin(), dragons.end());

    int count = 0;
    for (const auto& dragon : dragons) {
        int x = dragon.first;
        int y = dragon.second;

        if (x < s) {
            count++;
            s += y;
        } else break;
    }

    if (count == n) cout << "YES" << endl;
    else cout << "NO" << endl;

    return 0;
}